blake3-air/src/air.rs

```diff
@@ -0,0 +1,426 @@
+use core::borrow::Borrow;
+
+use itertools::izip;
+use p3_air::utils::{add2, add3, pack_bits_le, xor, xor_32_shift};
+use p3_air::{Air, AirBuilder, BaseAir};
+use p3_field::FieldAlgebra;
+use p3_matrix::Matrix;
+
+use crate::columns::{Blake3Cols, NUM_BLAKE3_COLS};
+use crate::constants::{permute, BITS_PER_LIMB, IV};
+use crate::{Blake3State, FullRound, QuarterRound};
+
+/// Assumes the field size is at least 16 bits.
+#[derive(Debug)]
+pub struct Blake3Air {}
+
+impl Blake3Air {
+    /// Verify that the mixing round function has been correctly computed.
+    ///
+    /// We assume that the values in a, b, c, d have all been range checked to be
+    /// either boolean (for b, d) or < 2^16 (for a, c). This both range checks all x', x''
+    /// and auxiliary variables as well as checking the relevant constraints between
+    /// them to conclude that the outputs are correct given the inputs.
+    fn mixing_function<AB: AirBuilder>(
```

Comment:
nit: I didn't really see "mixing function" in the specs, would it make sense to call `quarter_round`?

Reply:
Yeah that's a fair. They refer to it as a mixing function in their reference implementation: https://github.com/BLAKE3-team/BLAKE3/blob/master/reference_impl/reference_impl.rs.

The only issue with `quarter_round` is that its sorta a bad name as you do it `8` times per round. But I'll change it to that given that's what's in their docs.